agent/bench-scripts/pbench-fio

```diff
@@ -603,28 +630,28 @@ function fio_run_job() {
 
 	if [ ! -z "$clients" ]; then
 		debug_log "creating directories on the clients"
-		for client in `echo $clients | sed -e s/,/" "/g`; do
+		for client in "${client_names[@]}"; do
 			ssh $ssh_opts $client mkdir -p $benchmark_results_dir &
 		done
 		wait
 		debug_log "opening port 8765 on firewall on the clients"
-		for client in `echo $clients | sed -e s/,/" "/g`; do
-			ssh $ssh_opts $client "firewall-cmd --add-port=$fio_server_port/tcp >/dev/null" &
+		for client in "${client_names[@]}"; do
+			ssh $ssh_opts $client "firewall-cmd --add-port=${client_ports[$client]:-$fio_server_port}/tcp >/dev/null" &
 		done
 		wait
 		debug_log "killing any old fio process on the clients"
-		for client in `echo $clients | sed -e s/,/" "/g`; do
+		for client in "${client_names[@]}"; do
 			ssh $ssh_opts $client "killall fio >/dev/null 2>&1" &
 		done
 		wait
 		debug_log "starting new fio process on the clients"
-		for client in `echo $clients | sed -e s/,/" "/g`; do
-			ssh $ssh_opts $client "pushd $benchmark_results_dir >/dev/null; screen -dmS fio-server bash -c ''$benchmark_bin' --server 2>&1 >client-result.txt'"
+		for client in "${client_names[@]}"; do
+			ssh $ssh_opts $client "pushd $benchmark_results_dir >/dev/null; screen -dmS fio-server bash -c ''$benchmark_bin' --server${client_ports[$client]:+=,${client_ports[$client]}} 2>&1 >client-result.txt'"
```

Comment:
IIUC, if the array entry exists and has value say 1234, then this evaluates to `--server=,1234` -  why is the comma present? Should it be there?

Reply:
According to the [documentation](https://fio.readthedocs.io/en/latest/fio_doc.html#client-server), that `--server=,1234` means, "Start a fio server, listening on all interfaces on port 1234."

Reply:
Yes, this is correct.

Reply:
OK, thanks!